### PR TITLE
Pool creation fee

### PR DESCRIPTION
**Pull request by antstalepresh**

### CODE REVIEW

File: x/pool-incentives/keeper/keeper_test.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func (suite *KeeperTestSuite) preparePoolWithPoolParams(poolParams gammtypes.Poo
 			suite.ctx,
 			acc,
 			sdk.NewCoins(
+				sdk.NewCoin("uosmo", sdk.NewInt(10000000000)),
```
Comment (ValarDragon) [inline x/pool-incentives/keeper/keeper_test.go:55]: This is just a drive-by change right? Didn't really affect anything relative to this PR AFAICT. (If so, thats fine)

File: x/gamm/types/params.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if v.Validate() != nil {
+		return fmt.Errorf("invalid pool creation fee: %+v", i)
+	}
+
+	if v.IsZero() {
+		return fmt.Errorf("zero pool creation fee is not allowed: %+v", i)
+	}
```
Comment (ValarDragon) [inline x/gamm/types/params.go:63]: Can we allow a `0` validation fee?

File: x/gamm/types/params.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// Implements params.ParamSet
+func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
+	return paramtypes.ParamSetPairs{
+		paramtypes.NewParamSetPair(KeyPoolCreationFee, &p.PoolCreationFee, validatePoolCreationFee),
+	}
+}
+
+func validatePoolCreationFee(i interface{}) error {
```
Comment (ValarDragon) [inline x/gamm/types/params.go:51]: Inspired the deposit bug, we should also add checking that the new balance is a balance that actually exists on chain. We can do that later though.

**Review by ValarDragon — APPROVED** [review]
LGTM! Only question is that can we easily allow 0 fee. If not, its fine

File: x/gamm/types/params.go (hunk 2 above)
Comment (ValarDragon) [reply x/gamm/types/params.go:63]: (Like would we run into an error somewhere with coins.Valid() or sdk.NewCoins

Comment (antstalepresh) [reply x/gamm/types/params.go:63]: Will allow zero fee

File: x/gamm/types/params.go (hunk 3 above)
Comment (antstalepresh) [reply x/gamm/types/params.go:51]: This is basic validation - and it's not using the kvstore - for additional check can set the validation function on SetParams probably.

File: x/gamm/types/params.go (hunk 2 above)
Comment (antstalepresh) [reply x/gamm/types/params.go]: allowed empty fee

**Review by ValarDragon — APPROVED** [review]
Great job on this! LGTM
